### DISCUSSION TITLE
Convert hostname/fqdn to IP address

### DISCUSSION THREAD

**Discussion by amarnathpv** [vectordotdev/vector#18772] [General]
Is there a way to convert hostname / FQDN to IP address as part of remap?
I noticed reverse_dns, but that effectively gets to the FQDN, but looking for IP address.
TIA.

**Comment by jszwedko** [comment]
Do you mean do a DNS lookup for, say, an A record? Or something else? This functionality doesn't currently exist in Vector, though you could do it with Lua. Do you mind opening a feature request with more details about your use-case?

**Comment by amarnathpv** [comment]
Yes. To elaborate more, the hostname extracted from Syslog feed requires to enrich with IP translated for the sink consumer.
Do you have any examples to point me towards Lua?

On other hand, would be nice to have this feature given there are other IP based functions. I will raise a FR! Rest caveats with DNS etc., falls to the function for reverse_dns.

TIA.
